### PR TITLE
Dev heppyboost

### PR DESCRIPTION
**Pull request by gkasieczka**
Adds substructure variables (ungroomed fj, nsubjettiness, trimmed fj, MultiR-HTT)
@arizzi

### CODE REVIEW

File: VHbbAnalysis/Heppy/test/vhbb.py [hunk 1]
```diff
@@ -64,6 +64,13 @@
                 "ajidxCSV"       : NTupleCollection("aJCidx",    objectInt, 2,help="additional jet indices CSV"),
                 "hjidx3cj"       : NTupleCollection("hJ3Cidx",    objectInt, 2,help="Higgs jet indices 3 cen jets"),
                 "ajidx3cj"       : NTupleCollection("aJ3Cidx",    objectInt, 2,help="additional jet indices 3 cen  jets"),
+
+                # Boosted analysis objects
+                "ungroomedFatjets"       : NTupleCollection("ungroomedFatjets", fatjetType, 10,help="CA, R=1.5, pT > 200 GeV, no grooming"),
```
Comment (arizzi) [inline VHbbAnalysis/Heppy/test/vhbb.py]: you should add those only when running a config that has the boost stuff in (otherwise it crashes)

Comment (gkasieczka) [inline VHbbAnalysis/Heppy/test/vhbb.py]: Thanks - fixed with new commit (c56bf8e)

**Comment by arizzi** [comment]
btw it seems to go not very fast...  are we running it on all events / all PTs or you are doing some kind of preselection in the cmsRun job ?

**Comment by gkasieczka** [comment]
Fatjets are required to have a pT>200 GeV.

I did a quick/dirty check of timing using
time python vhbb-combined.py
and commenting out different parts.

process      time (real)
btag+boost 100s
boost            65s
btag              90s
neither          54s

So boost seems to cost ~10s and btag around ~35s.

(Tested on Z'->ttbar with m(Z')=1 TeV so the amount of boosted jets is probably higher than in our average sample)

**Comment by arizzi** [comment]
yes but the b-tag thing is just for testing, not really meant to run it on
later productions

**Comment by gkasieczka** [comment]
I understand. What I meant that most of the slowdown when running vhbb-combined is not due to the substructure variables (I was rather surprised myself how much things slowed down when I added the b-tags)

**Comment by arizzi** [comment]
testing on ttbar, that is much more representative of our average samples,
I see

btag stuff:
TimeReport   0.000123   0.000129   0.000123   0.000129   0.000123
0.000129 ak4JetTracksAssociatorAtVertexPF
TimeReport   0.000874   0.000960   0.000874   0.000960   0.000874
0.000960 combinedInclusiveSecondaryVertexV2BJetTags
TimeReport   0.000459   0.000463   0.000459   0.000463   0.000459
0.000463 combinedSecondaryVertexBJetTags
TimeReport   0.006879   0.007088   0.006879   0.007088   0.006879
0.007088 impactParameterTagInfos
TimeReport   0.000334   0.000338   0.000334   0.000338   0.000334
0.000338 inclusiveSecondaryVertexFinderTagInfos
TimeReport   0.004259   0.004263   0.004259   0.004263   0.004259
0.004263 secondaryVertexTagInfos
TimeReport   0.000512   0.000519   0.000512   0.000519   0.000512
0.000519 unpackedTracksAndVertices

i.e. 13ms/ev for btag

substructures:
TimeReport   0.007077   0.007156   0.007077   0.007156   0.007077
0.007156 ca15PFJetsCHS
TimeReport   0.001257   0.001260   0.001257   0.001260   0.001257
0.001260 ca15PFJetsCHSNSubjettiness
TimeReport   0.006807   0.006812   0.006807   0.006812   0.006807
0.006812 ca15PFTrimmedJetsCHS
TimeReport   0.003812   0.003817   0.003812   0.003817   0.003812
0.003817 chs
TimeReport   0.006955   0.006960   0.006955   0.006960   0.006955
0.006960 looseMultiRHTT

i.e. 27ms/ev for substructures

**Comment by gkasieczka** [comment]
Thanks for checking! So this is  a similar order of magnitude but a bit smaller to what I observed in the Z': 10s/300events ~33ms/ev.

So the question is: would this fit in the time budget?

**Comment by arizzi** [comment]
well right now we are a bit overbudget adding the various VBF and ttH requests that are not really relevant for VH. In the worst case scenario we can keep the same codebase and have different productions for different channels (so that each would run only on the samples that really needs)...

The other little issue with this PR is the fact that needs to pull in an recompile additional packages, so I was wondering if we want to have a separate branch (but then it would be a pain to keep the two in sync)

**Pull request merged**